### PR TITLE
Bugfix: since and limitBytes unsupported

### PR DESCRIPTION
**Pull request by cwdsuzhou**
Fix https://github.com/virtual-kubelet/virtual-kubelet/issues/820

### CODE REVIEW

**Comment by cwdsuzhou** [comment]
ping @cpuguy83  PTAL

**Comment by cpuguy83** [comment]
This should go into github.com/virtual-kubelet/node-cli
The cmd in this repo is only used for e2e tests.

Thanks!

**Comment by cpuguy83** [comment]
Argh... wrong tab....

**Comment by cpuguy83** [comment]
Can we use what kubelet is doing?

```go
	query := request.Request.URL.Query()
	// backwards compatibility for the "tail" query parameter
	if tail := request.QueryParameter("tail"); len(tail) > 0 {
		query["tailLines"] = []string{tail}
		// "all" is the same as omitting tail
		if tail == "all" {
			delete(query, "tailLines")
		}
	}
	// container logs on the kubelet are locked to the v1 API version of PodLogOptions
	logOptions := &v1.PodLogOptions{}
	if err := legacyscheme.ParameterCodec.DecodeParameters(query, v1.SchemeGroupVersion, logOptions); err != nil {
		response.WriteError(http.StatusBadRequest, fmt.Errorf(`{"message": "Unable to decode query."}`))
		return
	}
	logOptions.TypeMeta = metav1.TypeMeta{}
	if errs := validation.ValidatePodLogOptions(logOptions); len(errs) > 0 {
		response.WriteError(http.StatusUnprocessableEntity, fmt.Errorf(`{"message": "Invalid request."}`))
		return
	}
```

**Comment by cpuguy83** [comment]
I should clarify, I don't want to pull in a bunch of k8s packages do accomplish this (b/c this tends to be an insane amount of code), but I do want to make sure we are decoding everything we should be and with proper API versioning.

**Comment by cwdsuzhou** [comment]
> I should clarify, I don't want to pull in a bunch of k8s packages do accomplish this (b/c this tends to be an insane amount of code), but I do want to make sure we are decoding everything we should be and with proper API versioning.

Got it! But we have exposed `ContainerLogOpts` to  users, so if we want to support `PodLogOptions`, we have to adapt to it.

**Comment by cwdsuzhou** [comment]
done @cpuguy83

File: node/api/logs.go [hunk 1]
```diff
@@ -39,6 +44,10 @@ type ContainerLogOpts struct {
 	Timestamps bool
 }
 
+func init() {
+	legacyscheme.Scheme.AddKnownTypes(v1.SchemeGroupVersion, &v1.PodLogOptions{})
```
Comment (cpuguy83) [inline node/api/logs.go:48]: Let's not use `func init()`.

Also is there a way to not use a singleton like this?

**Comment by cpuguy83** [comment]
This PR seems similar to #806
I do think I'd prefer to manually parse the request over pulling in more from k8s/api.

**Comment by cwdsuzhou** [comment]
> This PR seems similar to #806
> I do think I'd prefer to manually parse the request over pulling in more from k8s/api.

Seems similar, but I would like use the  existing codes to keep vk and kubelet have the same behaviors. If kubelet's behavior updates, vk can automatically update.

**Comment by cpuguy83** [comment]
I think I'd prefer #806 just because I don't want to pull in a bunch of packages from k8s.
In fact I'd like to reduce what we can that's not in the core types or client packages.

**Pull request closed without merging**